### ISSUE TITLE
[feature requested] : to not lost indentation  in code strings

### ISSUE THREAD

**Issue by philibe** [rstudio/rstudio#4437]
When I copy SQL or JS from an external notepad to paste RStudio IDE (Server or not), I lost the indentation.

What I have in a notepad and expected pasted in RStudio:
```SQL
sql <- "SELECT *
FROM employees
WHERE EXISTS (SELECT *
              FROM contacts
              WHERE employees.last_name = contacts.last_name
              AND employees.first_name = contacts.first_name);
"
```

```R
callback = JS(" 
                                table.column(1).nodes().to$().css({
                                  cursor: 'pointer'
                                });
                                var format = function (d) {
                                  return '<div style=\"background-color:#eee;color:black; padding: .5em;\">' +
                                  '<b> bla bla :&nbsp;</b>' +  d[15] +
                                  '</div>';
                                };
                                table.on('click', 'td.details-control', function () {
                                    var td = $(this),
                                    row = table.row(td.closest('tr'));
                                    if (row.child.isShown()) {
                                     row.child.hide();
                                     td.html('&oplus;');
                                   } else {
                                    row.child(format(row.data())).show();
                                    td.html('&CircleMinus;');
                                  }
                                });
                                "
                  )         
  ```        

But in RStudio, the indentation is lost when the code is pasted:

```SQL
sql <- "SELECT *
FROM employees
WHERE EXISTS (SELECT *
FROM contacts
WHERE employees.last_name = contacts.last_name
AND employees.first_name = contacts.first_name);
"
```

```R
callback = JS(" 
                                table.column(1).nodes().to$().css({
                                cursor: 'pointer'
                                });
                                var format = function (d) {
                                return '<div style=\"background-color:#eee;color:black; padding: .5em;\">' +
                                '<b> Lots / Fourn pour les MP:&nbsp;</b>' +  d[15] +
                                '</div>';
                                };
                                table.on('click', 'td.details-control', function () {
                                var td = $(this),
                                row = table.row(td.closest('tr'));
                                if (row.child.isShown()) {
                                row.child.hide();
                                td.html('&oplus;');
                                } else {
                                row.child(format(row.data())).show();
                                td.html('&CircleMinus;');
                                }
                                });
                                "
                  )         
  ```        

Thank in advance :)

__edit:__ and in a new comment, an addendum to my issue

### System details

```
RStudio Edition : RStudio Server (x86_64-pc-linux-gnu (64-bit) 
Running under: Ubuntu 16.04.4 LTS)
RStudio Version : 1.1.419
OS Version      : Windows 7 / Firefox 64.0.2
R Version       : R version 3.4.4 
```

<details><summary>Session_info</summary>

``` 
> sessionInfo()
>  sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.4 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] mvbutils_2.8.232      tidyverse_1.2.1       magrittr_1.5          purrr_0.2.4           gtools_3.5.0         
 [6] devtools_1.13.6       glue_1.3.0            odbc_1.1.6            shinyjs_1.0           collapsibleTree_0.1.6
[11] dplyr_0.7.8           statnet.common_4.0.0  xtable_1.8-3          shinydashboard_0.7.0  DT_0.4               
[16] shinyRGL_0.1.0        rgl_0.95.1441         RSQLite_2.1.1         reprex_0.2.1          shiny_1.1.0          
[21] bindrcpp_0.2.2        DBI_1.0.0             taRifx_1.0.6.1        forcats_0.3.0         stringr_1.3.1        
[26] readr_1.1.1           tidyr_0.8.2           tibble_1.4.2         

loaded via a namespace (and not attached):
 [1] nlme_3.1-137       fs_1.2.6           lubridate_1.7.4    bit64_0.9-7        RColorBrewer_1.1-2 httr_1.3.1        
 [7] data.tree_0.7.5    tools_3.4.4        R6_2.3.0           lazyeval_0.2.1     colorspace_1.3-2   withr_2.1.2       
[13] tidyselect_0.2.5   gridExtra_2.3      mnormt_1.5-5       bit_1.1-14         compiler_3.4.4     cli_1.0.0         
[19] rvest_0.3.2        xml2_1.2.0         influenceR_0.1.0   scales_1.0.0       psych_1.8.4        digest_0.6.17     
[25] foreign_0.8-70     pkgconfig_2.0.2    htmltools_0.3.6    htmlwidgets_1.3    rlang_0.3.1        readxl_1.1.0      
[31] rstudioapi_0.8     bindr_0.1.1        visNetwork_2.0.3   jsonlite_1.6       rgexf_0.15.3       Rcpp_1.0.0        
[37] munsell_0.5.0      viridis_0.5.1      yaml_2.2.0         stringi_1.2.2      plyr_1.8.4         grid_3.4.4        
[43] blob_1.1.1         parallel_3.4.4     promises_1.0.1     crayon_1.3.4       lattice_0.20-35    haven_1.1.2       
[49] hms_0.4.2          pillar_1.2.2       igraph_1.2.1       reshape2_1.4.3     XML_3.98-1.11      downloader_0.4    
[55] modelr_0.1.2       httpuv_1.4.3.9002  cellranger_1.1.0   gtable_0.2.0       assertthat_0.2.0   ggplot2_2.2.1     
[61] mime_0.5           broom_0.4.4        later_0.7.5.9001   viridisLite_0.3.0  memoise_1.1.0      Rook_1.1-1        
[67] DiagrammeR_1.0.0   brew_1.0-6        
```

</details>

**Comment by pascaloettli** [comment]
I can copy/paste your indented code in RStudio (R Script) without losing the format (Ubuntu 18.04, RStudio Version 1.2.1321). So please edit your message to include the template you see when opening an issue, i.e.  
### System details

    RStudio Edition : 
    RStudio Version : 
    OS Version      : 
    R Version       :

**Comment by philibe** [comment]
addendum to my issue

When I do "File / New File / RScript" and copy from notepad.exe  to RStudio Server:

copy / paste: OK
```
js <- "if (a==1) {
         b=2
      }"
```

copy/ paste: indentation lost (note the line feed in the string )
```
js <- "
if (a==1) {
  b=2
}
"
       
copy / paste: indentation lost (note the line feed before the assgination code)
```
   
js <- "if (a==1) {
         b=2
      }"
```

Note that this issue occurs also with others notepad and RStudio Desktop in Windows. Or with Unbuntu text editor and RStudio Server Ubuntu. Same RStudio version.

**Comment by ronblum** [comment]
@philibe Thank you for reporting the issue and providing your environment information. Are you able to try out [the latest RStudio 1.2 preview](https://www.rstudio.com/products/rstudio/download/preview/) to see if you can reproduce it there?

And thank you @pascaloettli for so often looking at issues and providing input! We count on all of you to improve and expand our products.

**Comment by david-cortes** [comment]
I recall this problem is already fixed in the latest builds: https://github.com/rstudio/rstudio/issues/3036

**Comment by philibe** [comment]
If I read correctly git commit it's therefore fixed  in v1.2.1320, i.e. in 1.2 preview.  In others words: not in my 1.1.x version. So I wait this >=1.2.13.20 version :)

**Comment by pascaloettli** [comment]
@philibe Use the Ubuntu14/Debian8 version. An yes, the next release of RStudio will be 1.2, so people are mostly working on this release. 1.1 will be deprecated

**Comment by philibe** [comment]
@pascaloettli

I am in a very little IT service: I have not a new server computer for test an Ubuntu software with incorrect version, if I test it will be on my development server with good version. :)

**Comment by pascaloettli** [comment]
@philibe  This is the version you have to use. There is no Rstudio Server version compiled for each Ubuntu release (if I am not mistaken.)

**Comment by philibe** [comment]
The current issue, copy/paste indentation, seems fixed for RStudio Server 1.2.1322 :)

with https://s3.amazonaws.com/rstudio-ide-build/server/trusty/amd64/rstudio-server-1.2.1322-amd64.deb

**Comment by ronblum** [comment]
Thank you for confirming!
